docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -470,13 +470,23 @@ An unused bean:
 * does not declare any producer which is eligible for injection to any injection point,
 * is not directly eligible for injection into any `javax.enterprise.inject.Instance` or `javax.inject.Provider` injection point
 
+This optimization applies to all forms of bean declarations: bean class, producer method, producer field.
+
 Users can instruct the container to not remove any of their specific beans (even if they satisfy all the rules specified above) by annotating them with `io.quarkus.arc.Unremovable`.
+This annotation can be placed on the type declaration, on a producer method or on a producer field.
```

Comment:
The repetition of "on" in this sentence rings false in my ears. I am not a native speaker though; I could be wrong :)
How about we just say the following?
`This annotation can be placed on types, methods and fields.`

Reply:
Yes but with the producer qualifier
```
This annotation can be placed on types, producer methods, and producer fields.
```

Reply:
Fair enough

Reply:
done